In [1]:
import pandas as pd

In [2]:
df= pd.read_csv('titanic.csv')

In [6]:
df.head(50)

PassengerId  Survived  Pclass  \
0             1         0       3   
1             2         1       1   
2             3         1       3   
3             4         1       1   
4             5         0       3   
5             6         0       3   
6             7         0       1   
7             8         0       3   
8             9         1       3   
9            10         1       2   
10           11         1       3   
11           12         1       1   
12           13         0       3   
13           14         0       3   
14           15         0       3   
15           16         1       2   
16           17         0       3   
17           18         1       2   
18           19         0       3   
19           20         1       3   
20           21         0       2   
21           22         1       2   
22           23         1       3   
23           24         1       1   
24           25         0       3   
25           26         1       3   
26           27         0       3   
27           28         0       1   
28           29         1       3   
29           30         0       3   
30           31         0       1   
31           32         1       1   
32           33         1       3   
33           34         0       2   
34           35         0       1   
35           36         0       1   
36           37         1       3   
37           38         0       3   
38           39         0       3   
39           40         1       3   
40           41         0       3   
41           42         0       2   
42           43         0       3   
43           44         1       2   
44           45         1       3   
45           46         0       3   
46           47         0       3   
47           48         1       3   
48           49         0       3   
49           50         0       3   

                                                 Name     Sex   Age  SibSp  \
0                             Braund, Mr. Owen Harris    male  22.0      1   
1   Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                              Heikkinen, Miss. Laina  female  26.0      0   
3        Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                            Allen, Mr. William Henry    male  35.0      0   
5                                    Moran, Mr. James    male   NaN      0   
6                             McCarthy, Mr. Timothy J    male  54.0      0   
7                      Palsson, Master. Gosta Leonard    male   2.0      3   
8   Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                 Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   
10                    Sandstrom, Miss. Marguerite Rut  female   4.0      1   
11                           Bonnell, Miss. Elizabeth  female  58.0      0   
12                     Saundercock, Mr. William Henry    male  20.0      0   
13                        Andersson, Mr. Anders Johan    male  39.0      1   
14               Vestrom, Miss. Hulda Amanda Adolfina  female  14.0      0   
15                   Hewlett, Mrs. (Mary D Kingcome)   female  55.0      0   
16                               Rice, Master. Eugene    male   2.0      4   
17                       Williams, Mr. Charles Eugene    male   NaN      0   
18  Vander Planke, Mrs. Julius (Emelia Maria Vande...  female  31.0      1   
19                            Masselmani, Mrs. Fatima  female   NaN      0   
20                               Fynney, Mr. Joseph J    male  35.0      0   
21                              Beesley, Mr. Lawrence    male  34.0      0   
22                        McGowan, Miss. Anna "Annie"  female  15.0      0   
23                       Sloper, Mr. William Thompson    male  28.0      0   
24                      Palsson, Miss. Torborg Danira  female   8.0      3   
25  Asplund, Mrs. Carl Oscar (Selma Augusta Emilia...  female  38.0      1   
26        

In [17]:
df_new= df.drop(['Name','PassengerId','Ticket','Parch','Cabin','Embarked','SibSp'],axis='columns')

Did some manipulations on the dataset to perform analysis productively

In [18]:
df_new.head(25)

Survived  Pclass     Sex   Age     Fare
0          0       3    male  22.0   7.2500
1          1       1  female  38.0  71.2833
2          1       3  female  26.0   7.9250
3          1       1  female  35.0  53.1000
4          0       3    male  35.0   8.0500
5          0       3    male   NaN   8.4583
6          0       1    male  54.0  51.8625
7          0       3    male   2.0  21.0750
8          1       3  female  27.0  11.1333
9          1       2  female  14.0  30.0708
10         1       3  female   4.0  16.7000
11         1       1  female  58.0  26.5500
12         0       3    male  20.0   8.0500
13         0       3    male  39.0  31.2750
14         0       3  female  14.0   7.8542
15         1       2  female  55.0  16.0000
16         0       3    male   2.0  29.1250
17         1       2    male   NaN  13.0000
18         0       3  female  31.0  18.0000
19         1       3  female   NaN   7.2250
20         0       2    male  35.0  26.0000
21         1       2    male  34.0  13.0000
22         1       3  female  15.0   8.0292
23         1       1    male  28.0  35.5000
24         0       3  female   8.0  21.0750

In [19]:
import numpy as np

In [25]:
Avg_age=df_new.Age.median()
Avg_age

28.0

In [26]:
df_new.Age=df_new.Age.fillna(Avg_age)
df_new.head(25)

Survived  Pclass     Sex   Age     Fare
0          0       3    male  22.0   7.2500
1          1       1  female  38.0  71.2833
2          1       3  female  26.0   7.9250
3          1       1  female  35.0  53.1000
4          0       3    male  35.0   8.0500
5          0       3    male  28.0   8.4583
6          0       1    male  54.0  51.8625
7          0       3    male   2.0  21.0750
8          1       3  female  27.0  11.1333
9          1       2  female  14.0  30.0708
10         1       3  female   4.0  16.7000
11         1       1  female  58.0  26.5500
12         0       3    male  20.0   8.0500
13         0       3    male  39.0  31.2750
14         0       3  female  14.0   7.8542
15         1       2  female  55.0  16.0000
16         0       3    male   2.0  29.1250
17         1       2    male  28.0  13.0000
18         0       3  female  31.0  18.0000
19         1       3  female  28.0   7.2250
20         0       2    male  35.0  26.0000
21         1       2    male  34.0  13.0000
22         1       3  female  15.0   8.0292
23         1       1    male  28.0  35.5000
24         0       3  female   8.0  21.0750

In [29]:
inputs= df_new.drop('Survived',axis='columns')
target= df_new['Survived']

In [31]:
target.head(5)

0    0
1    1
2    1
3    1
4    0
Name: Survived, dtype: int64

In [33]:
inputs.head(5)

Pclass     Sex   Age     Fare
0       3    male  22.0   7.2500
1       1  female  38.0  71.2833
2       3  female  26.0   7.9250
3       1  female  35.0  53.1000
4       3    male  35.0   8.0500

In [34]:
from sklearn.preprocessing import LabelEncoder

In [35]:
Encoded_Sex= LabelEncoder()

In [39]:
inputs['Gender']= Encoded_Sex.fit_transform(inputs['Sex'])
inputs.head(5)

Pclass     Sex   Age     Fare  Gender
0       3    male  22.0   7.2500       1
1       1  female  38.0  71.2833       0
2       3  female  26.0   7.9250       0
3       1  female  35.0  53.1000       0
4       3    male  35.0   8.0500       1

In [50]:
inputs_rev=inputs.drop('Sex',axis='columns')

In [43]:
from sklearn import tree

In [44]:
model= tree.DecisionTreeClassifier()

In [45]:
from sklearn.model_selection import train_test_split

In [51]:
X_train,X_test,y_train,y_test= train_test_split(inputs_rev,target,test_size=0.2)

In [52]:
model.fit(X_train,y_train)

DecisionTreeClassifier()

In [55]:
y_test.head

<bound method NDFrame.head of 357    0
815    0
786    1
110    0
491    0
      ..
221    0
392    0
98     1
274    1
714    0
Name: Survived, Length: 179, dtype: int64>

In [56]:
X_test

Pclass   Age     Fare  Gender
357       2  38.0  13.0000       0
815       1  28.0   0.0000       1
786       3  18.0   7.4958       0
110       1  47.0  52.0000       1
491       3  21.0   7.2500       1
..      ...   ...      ...     ...
221       2  27.0  13.0000       1
392       3  28.0   7.9250       1
98        2  34.0  23.0000       0
274       3  28.0   7.7500       0
714       2  52.0  13.0000       1

[179 rows x 4 columns]

In [57]:
model.predict(X_test)

array([1, 0, 1, 1, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1,
       1, 0, 1, 0, 1, 1, 0, 1, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 1, 1, 1,
       0, 0, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 0,
       1, 0, 1, 1, 1, 0, 1, 1, 1, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 1, 0, 1,
       0, 1, 1, 0, 1, 1, 0, 0, 1, 0, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0,
       0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0,
       1, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 1,
       0, 0, 0, 1, 1, 0, 1, 0, 1, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0,
       1, 1, 0], dtype=int64)

In [60]:
model.score(inputs_rev,target)

0.941638608305275

Score is 0.9416, which means that accuracy of our trained model is 94.16%.